### PR TITLE
TEST: Rewrite GeoPointParsingTests#testEqualsHashCodeContract

### LINKED ISSUE #27633: [CI] GeoPointParsingTests#testEqualsHashCodeContract failed
I can reproduce this locally.

```
gradle :core:test 
-Dtests.seed=CCFED399964CE6C0 
-Dtests.class=org.elasticsearch.index.search.geo.GeoPointParsingTests 
-Dtests.method="testEqualsHashCodeContract" 
-Dtests.security.manager=true 
-Dtests.locale=nl-BE 
-Dtests.timezone=Canada/Central
```

### PR DESCRIPTION
**Pull request by dnhatn**
The hashCode contract states that equal objects must have equal hash codes, however the unequal objects are not required to have unequal hashCodes. This commit rewrites `GeoPointParsingTests#testEqualsHashCodeContract` using `#checkEqualsAndHashCode` helper.

Closes #27633

### CODE REVIEW

File: core/src/test/java/org/elasticsearch/index/search/geo/GeoPointParsingTests.java [hunk 1]
```diff
@@ -77,12 +78,10 @@ public void testEqualsHashCodeContract() {
...
         // symmetry
-        assertTrue(x.hashCode() == y.hashCode());
+        assertThat(x.hashCode(), equalTo(y.hashCode()));
         // transitivity
-        assertTrue(y.hashCode() == z.hashCode());
-        assertTrue(x.hashCode() == z.hashCode());
-        // inequality
-        assertFalse(x.hashCode() == a.hashCode());
+        assertThat(y.hashCode(), equalTo(z.hashCode()));
+        assertThat(x.hashCode(), equalTo(z.hashCode()));
```
Comment (jasontedor) [inline core/src/test/java/org/elasticsearch/index/search/geo/GeoPointParsingTests.java]: There is `EqualsHashCodeTestUtils#checkEqualsAndHashCode`; maybe this test should be rewritten to use that?

**Review by jasontedor — COMMENTED** [review]
I left a comment.

File: core/src/test/java/org/elasticsearch/index/search/geo/GeoPointParsingTests.java (hunk 1 above)
Comment (dnhatn) [reply core/src/test/java/org/elasticsearch/index/search/geo/GeoPointParsingTests.java]: Sure, I will rewrite this test. Thank you for your suggestion :).

**Comment by dnhatn** [comment]
@jasontedor, I've rewritten the test using `#checkEqualsAndHashCode` helper. Could you please take another look? Thank you.

File: core/src/test/java/org/elasticsearch/index/search/geo/GeoPointParsingTests.java [hunk 2]
```diff
@@ -57,32 +59,18 @@ public void testGeoPointReset() throws IOException {
...
-        assertTrue(x.hashCode() == y.hashCode());
-        // transitivity
-        assertTrue(y.hashCode() == z.hashCode());
-        assertTrue(x.hashCode() == z.hashCode());
-        // inequality
-        assertFalse(x.hashCode() == a.hashCode());
+        // GeoPoint doesn't care about coordinate system bounds, this simply validates equality and hashCode.
+        final DoubleSupplier randomDelta = () -> randomValueOtherThan(0.0, () -> randomDoubleBetween(-1000000, 1000000, true));
+        checkEqualsAndHashCode(RandomGeoGenerator.randomPoint(random()), GeoPoint::new, pt -> {
+            switch (randomInt(2)) {
```
Comment (jasontedor) [inline core/src/test/java/org/elasticsearch/index/search/geo/GeoPointParsingTests.java]: I think this is an unnecessary use of randomization in a test. I think randomizing in situations like this only makes tests harder to debug. Why not simply test all the cases, we can enumerate them?

**Review by jasontedor — CHANGES_REQUESTED** [review]
Thanks for rewriting the test. I left a comment about the use of randomization.

File: core/src/test/java/org/elasticsearch/index/search/geo/GeoPointParsingTests.java (hunk 2 above)
Comment (dnhatn) [reply core/src/test/java/org/elasticsearch/index/search/geo/GeoPointParsingTests.java]: I pushed https://github.com/elastic/elasticsearch/pull/27634/commits/8109cafd77d9830eeca465261f30c3158abc4137.

**Review by jasontedor — APPROVED** [review]
LGTM.

**Comment by dnhatn** [comment]
Thanks @jasontedor.

**Pull request merged**